src/implot_visualization/app/main.cpp

```diff
@@ -140,91 +157,43 @@ static void main_loop(void *arg) {
-                ImPlot::SetupAxisLimits(ImAxis_X1, currentTime - 10.0, currentTime, ImGuiCond_Always);
-                ImPlot::SetupAxisScale(ImAxis_X1, ImPlotScale_Time);
-                ImPlot::SetupAxis(ImAxis_Y2, "phi(deg)", ImPlotAxisFlags_AuxDefault);
-                ImPlot::SetNextFillStyle(IMPLOT_AUTO_COL, 0.5f);
-                // if (buffer.Data.size() > 0) {
-                //     // ImPlot::PlotLine("Sinus", &buffer.Data[0].x, &buffer.Data[0].y, buffer.Data.size(), 0, buffer.Offset, 2 * sizeof(double));
-                // }
+                plotter.plotPower(subscriptions[1].signals);
                 ImPlot::EndPlot();
             }
 
             // Mains Frequency Plot
             if (ImPlot::BeginPlot("Mains Frequency")) {
-                static ImPlotAxisFlags xflags = ImPlotAxisFlags_None;
-                static ImPlotAxisFlags yflags = ImPlotAxisFlags_AutoFit | ImPlotAxisFlags_RangeFit;
-                ImPlot::SetupAxes("time (s)", "Frequency (Hz)", xflags, yflags);
-                auto   clock       = std::chrono::system_clock::now();
-                double currentTime = (std::chrono::duration_cast<std::chrono::milliseconds>(clock.time_since_epoch()).count()) / 1000.0;
-                ImPlot::SetupAxisLimits(ImAxis_X1, currentTime - 10.0, currentTime, ImGuiCond_Always);
-                ImPlot::SetupAxisScale(ImAxis_X1, ImPlotScale_Time);
-                ImPlot::SetNextFillStyle(IMPLOT_AUTO_COL, 0.5f);
-                // if (buffer.Data.size() > 0) {
-                //     // ImPlot::PlotLine("Sinus", &buffer.Data[0].x, &buffer.Data[0].y, buffer.Data.size(), 0, buffer.Offset, 2 * sizeof(double));
-                // }
+                plotter.plotMainsFrequency(subscriptions[1].signals);
                 ImPlot::EndPlot();
             }
         }
         ImPlot::EndSubplots();
 
         // Power Spectrum
         if (ImPlot::BeginPlot("Power Spectrum")) {
-            static ImPlotAxisFlags xflags = ImPlotAxisFlags_None;
-            static ImPlotAxisFlags yflags = ImPlotAxisFlags_AutoFit | ImPlotAxisFlags_RangeFit;
-            ImPlot::SetupAxes("Frequency (Hz)", "Power Density (dB)", xflags, yflags);
-            ImPlot::SetupAxisLimits(ImAxis_X1, 0, 7, ImGuiCond_Always);
-            ImPlot::SetNextFillStyle(IMPLOT_AUTO_COL, 0.5f);
-            // if (buffer.Data.size() > 0) {
-            //     // ImPlot::PlotLine("Sinus", &buffer.Data[0].x, &buffer.Data[0].y, buffer.Data.size(), 0, buffer.Offset, 2 * sizeof(double));
-            // }
+            plotter.plotPowerSpectrum(subscriptions[1].signals);
             ImPlot::EndPlot();
         }
         ImGui::End();
     }
 
     // Visualize Counter
     if (visualize_counter) {
-        fetch("http://localhost:8080/counter/testCounter", signals, &deserializer);
+        FetchUtils   fetchCounterSignals("http://localhost:8080/counter/testCounter", 1);
```

Comment:
This creates a new FetchUtils instance in each main_loop iteration -> the `signals` field the json gets deserialized into another instance than the one that is plotted in subsequent runs of the loop. Still have to test it, but if I'm not mistaken this will probably not work. But since the counter implementation is not needed in the final project, I think it is ok to remove it. If you want to keep it for testing it should probably be static.

Reply:
I agree that we can remove the counter visualization from the UI part. It was just implemented for first experience and leaving it in now would lead to more maintenance work than it brings benefit.